durabletask/internal/grpc_interceptor.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from collections import namedtuple
+from typing import List, Tuple
+
+import grpc
+
+
+class _ClientCallDetails(
+        namedtuple(
+            '_ClientCallDetails',
+            ['method', 'timeout', 'metadata', 'credentials', 'wait_for_ready', 'compression']),
+        grpc.ClientCallDetails):
+    """This is an implementation of the ClientCallDetails interface needed for interceptors.
+    This class takes six named values and inherits the ClientCallDetails from grpc package.
+    This class encloses the values that describe a RPC to be invoked.
+    """
+    pass
+
+def _intercept_call(
+            client_call_details: _ClientCallDetails, metadata: List[Tuple[str, str]]) -> _ClientCallDetails:
+    """Internal intercept_call implementation which adds metadata to grpc metadata in the RPC
+        call details."""
+    new_metadata = []
+    new_client_call_details = client_call_details
+    if client_call_details.metadata is not None:
+        new_metadata = list(metadata)
+    if metadata is not None:
+        new_metadata.extend(metadata)
+        new_client_call_details = _ClientCallDetails(
```

Comment:
Why do we need to create a new copy of the `_ClientCallDetails` object? Can't we mutate the `client_call_details` variable that's passed in?

Reply:
Please refer https://github.com/microsoft/durabletask-python/pull/16#discussion_r1330163140